src/tests/tests/storage.js

```diff
@@ -76,11 +76,28 @@
       {dnt: false, heuristicAction: "", nextUpdateTime: 100, userAction: ""});
     assert.ok(action_map.getItem('testsite.com').nextUpdateTime === 100);
 
-    let newValue = {dnt: false, heuristicAction: "", nextUpdateTime: 101, userAction: ""};
+    let newValue = {dnt: true, heuristicAction: constants.BLOCK, nextUpdateTime: 99, userAction: constants.USER_BLOCK};
     action_map.merge({'testsite.com': newValue});
 
-    // Merged in object should have new nextUpdateTime value
+    // userAction should be merged
+    assert.ok(action_map.getItem('testsite.com').userAction === constants.USER_BLOCK);
+
+    // heuristicAction should not
+    assert.ok(action_map.getItem('testsite.com').heuristicAction === "");
+
+    // DNT should not be merged if nextUpdateTime is earlier
+    assert.ok(action_map.getItem('testsite.com').nextUpdateTime === 100);
+    assert.ok(action_map.getItem('testsite.com').dnt === false);
```

Comment:
`assert.notOk` would be better here. QUnit assertions: https://api.qunitjs.com/assert/